pkg/controller/sub_controller/fe/prepare_modify.go

```diff
@@ -43,21 +43,18 @@ func (fc *Controller) prepareStatefulsetApply(ctx context.Context, cluster *v1.D
 	if cluster.Spec.FeSpec.Replicas == nil {
 		cluster.Spec.FeSpec.Replicas = resource.GetInt32Pointer(0)
 	}
-	scaleNumber := *(cluster.Spec.FeSpec.Replicas) - *(oldSt.Spec.Replicas)
-	// fe scale
-	/*	if scaleNumber != 0 { // set fe Phase as SCALING
-		cluster.Status.FEStatus.ComponentCondition.Phase = meta_v1.Scaling
-		// In Reconcile, it is possible that the status cannot be updated in time,
-		// resulting in an error in the status judgment based on the last status,
-		// so the status will be forced to modify here
-			if err := k8s.SetDorisClusterPhase(ctx, fc.K8sclient, cluster.Name, cluster.Namespace, meta_v1.Scaling, meta_v1.Component_FE); err != nil {
-			klog.Errorf("SetDorisClusterPhase 'SCALING' failed err:%s ", err.Error())
-			return err
-		}
-	}*/
 
-	if scaleNumber < 0 {
-		if err := fc.dropObserverFromSqlClient(ctx, fc.K8sclient, cluster); err != nil {
+	if *(cluster.Spec.FeSpec.Replicas) < *(cluster.Spec.FeSpec.ElectionNumber) {
+		fc.K8srecorder.Event(cluster, string(sc.EventWarning), string(sc.FESpecSetError), "The number of fe ElectionNumber is large than Replicas, Replicas has been corrected to the correct minimum value")
+		klog.Errorf("prepareStatefulsetApply namespace=%s,name=%s ,The number of fe ElectionNumber(%d) is large than Replicas(%d)", cluster.Namespace, cluster.Name, *(cluster.Spec.FeSpec.ElectionNumber), *(cluster.Spec.FeSpec.Replicas))
+		ele := *(cluster.Spec.FeSpec.ElectionNumber)
+		cluster.Spec.FeSpec.Replicas = &ele
+	}
+
+	willRemovedAmount := *(cluster.Spec.FeSpec.Replicas) - *(oldSt.Spec.Replicas)
```

Comment:
the real meaning is the opposite of removedAmount, so, the name should is willRemovedOppositeAmount shortly as wroa and add comment.